component/loki/source/windowsevent/arguments.go

```diff
@@ -0,0 +1,46 @@
+package windowsevent
+
+import (
+	"time"
+
+	"github.com/grafana/agent/component/common/loki"
+)
+
+// Arguments holds values which are used to configure the loki.source.windowsevent
+// component.
+type Arguments struct {
+	Locale               int                 `river:"locale,attr,optional"`
+	EventLogName         string              `river:"eventlog_name,attr,optional"`
+	XPathQuery           string              `river:"xpath_query,attr,optional"`
+	BookmarkPath         string              `river:"bookmark_path,attr,optional"`
+	PollInterval         time.Duration       `river:"poll_interval,attr,optional"`
+	ExcludeEventData     bool                `river:"exclude_event_data,attr,optional"`
+	ExcludeUserdata      bool                `river:"exclude_user_data,attr,optional"`
+	UseIncomingTimestamp bool                `river:"use_incoming_timestamp,attr,optional"`
+	ForwardTo            []loki.LogsReceiver `river:"forward_to,attr"`
+}
+
+func defaultArgs() Arguments {
+	return Arguments{
+		Locale:               0,
+		EventLogName:         "",
+		XPathQuery:           "*",
+		BookmarkPath:         "",
+		PollInterval:         3 * time.Second,
+		ExcludeEventData:     false,
+		ExcludeUserdata:      false,
+		UseIncomingTimestamp: false,
+	}
+}
+
+// UnmarshalRiver implements river.Unmarshaler.
+func (r *Arguments) UnmarshalRiver(f func(v interface{}) error) error {
+	*r = defaultArgs()
+
+	type arguments Arguments
+	if err := f((*arguments)(r)); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Should we be doing any validations here? (Such as making sure `eventlog_name` is provided if `xpath_query` uses the short form)

Reply:
I tried to do this a bit. 

The current implementation doesn't check and we could try to parse the field as XML but even with XML you may need the eventlog_name, so it felt more effort than it was worth. IMO we can ship it without and  add it later.